corehq/apps/data_interfaces/tasks.py

```diff
@@ -216,30 +216,26 @@ def delete_old_rule_submission_logs():
 
 
 @task(serializer='pickle')
-def task_operate_on_payloads(payload_ids, domain, action=''):
-    task = task_operate_on_payloads
-
-    if not payload_ids:
-        return {'messages': {'errors': [_('No Payloads are supplied')]}}
-
-    if not action:
-        return {'messages': {'errors': [_('No action specified')]}}
-
-    response = operate_on_payloads(payload_ids, domain, action, task)
-
-    return response
+def task_operate_on_payloads(record_ids, domain, action=''):
+    return operate_on_payloads(record_ids, domain, action,
+                               task=task_operate_on_payloads)
 
 
 @task(serializer='pickle')
-def task_generate_ids_and_operate_on_payloads(data, domain, action=''):
-    task = task_generate_ids_and_operate_on_payloads
+def task_generate_ids_and_operate_on_payloads(query_string_dict, domain, action=''):
```

Comment:
nit: IMO it'd be easier to follow if you passed `payload_id` and `repeater` (`repeater_id`?) as first-class parameters, rather than passing `request.POST` in.